krpc/msg.go

```diff
@@ -71,13 +75,13 @@ type Bep44Return struct {
 	Seq *int64      `bencode:"seq,omitempty"`
 }
 
-type Return struct {
+type Return[T CompactNodeInfo] struct {
 	// All returns are supposed to contain an ID, but what if they don't?
 	ID ID `bencode:"id"` // ID of the queried (and responding) node
 
 	// K closest nodes to the requested target. Included in responses to queries that imply
 	// traversal, for example get_peers, find_nodes, get, sample_infohashes.
-	Nodes  CompactIPv4NodeInfo `bencode:"nodes,omitempty"`
+	Nodes  T `bencode:"nodes,omitempty"`
```

Comment:
Rather than threading this type throughout, perhaps we should be passing its interpretation as a runtime parameter somewhere? I.e. pass CompactIPv4NodeInfo, or CompactI2PNodeInfo somewhere?